src/main/resources/static/api.yml

```diff
@@ -4238,6 +4238,38 @@ paths:
                 $ref: '_shared.yml#/components/schemas/Problem'
         500:
           $ref: '_shared.yml#/components/responses/500Response'
+  /users/me:
```

Comment:
we already have an endpoint to get current user's details on /profile (or /profile/v2 preferably).

we could add an optional param to that to indicate the user's details should be refreshed before being retrieved? I'm concerned that we're introducing an additional path to do something similar.

Reply:
The wrapper on /profile/v2 is so we can provide context back to the caller if the user doesn't have staff information so the UI can display something useful to the user when this occurs. The response still contains the same User type (which in your case will always be a TemporaryAccommodationUser)

If the UI calls /profile and there is no staff information the API will throw a 500 so the user will see 'something went wrong' which then leads to a manual support request when it could be handled with a useful message.

I believe CAS3 UI is currently calling /profile. So ideally, if you can, we should make this enhancement to /profile/v2 and also update the CAS3 UI to use this new version _and_ also handle what happens when no staff member exists e.g. where loadError=staff_record_not_found. An example of what CAS1 does is here - https://dsdmoj.atlassian.net/browse/APS-597. Rosie may know if there's an equivalent CAS3 ticket.